desc/objectives/_coils.py

```diff
@@ -0,0 +1,504 @@
+        this should also be set to True.
+        be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: Has no effect for this objective.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+    """
+
+    def __init__(
+        self,
+        coil,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        name=None,
+    ):
+        if target is None and bounds is None:
+            bounds = (0, 1)
+
+        super().__init__(
+            coil,
+            ["curvature"],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            grid=grid,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        super().build(use_jit=use_jit, verbose=verbose)
+
+        if self._normalize:
+            self._normalization = 1 / self._scales["a"]
+
+    def compute(self, params, constants=None):
+        """Compute coil curvature.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of the coil's degrees of freedom.
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self._constants.
+
+        Returns
+        -------
+        f : float or array of floats
+            Coil curvature.
+        """
+        data = super().compute(params, constants=constants)
+        data = tree_flatten(data, is_leaf=lambda x: isinstance(x, dict))[0]
+        out = jnp.concatenate([dat["curvature"] for dat in data])
+        return out
+
+
+class CoilTorsion(_CoilObjective):
+    """Coil torsion.
+
+    Parameters
+    ----------
+    coil : CoilSet or Coil
+        Coil(s) that are to be optimized
+    target : {float, ndarray}, optional
```

Comment:
the `{a, b}` in a type annotation means that it has to be one of those literal values. IE, in this case it means that `target` should be the literal python object `float`, rather than a number of type `float`. 

In this case, we just care that `target` has a particular type, so the notation would be `target : float, ndarray, optional`

see https://numpydoc.readthedocs.io/en/latest/format.html#parameters